source/hash_table.c

```diff
@@ -0,0 +1,582 @@
+/*
+* Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+// For more information on how the RH hash works and in particular how we do
+// deletions, see:
+// http://codecapsule.com/2013/11/17/robin-hood-hashing-backward-shift-deletion/
+
+#include <aws/common/hash_table.h>
+#include <aws/common/common.h>
+#include <aws/common/math.h>
+#include <aws/common/linked_list.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
```

Comment:
stdio only needed for debugging functions. Guard it with if 0? or include it in the debugging code section.